### PR TITLE
24w10a

### PR DESCRIPTION
**Pull request by Apollounknowndev**
Todo:

- [X] Banner patterns
- [X] Wolf variants
- [X] Components in recipes
- [X] Misc. item component changes
- [X] Slot predicate

### CODE REVIEW

File: java/1.20.5/src/schemas/Recipe.ts [hunk 1]
```diff
@@ -120,9 +120,19 @@ export function initRecipeSchemas(schemas: SchemaRegistry, collections: Collecti
...
+    default: () => ({
+      id: 'minecraft:stone'
+    })
+  }))
+
   schemas.register('recipe_result', Mod(ObjectNode({
     id: StringNode({ validator: 'resource', params: { pool: 'item' } }),
-    count: Opt(Mod(NumberNode({ integer: true }), { default: () => 1 }))
+    count: Opt(Mod(NumberNode({ integer: true }), { default: () => 1 })),
+    components: Opt(Reference('data_component_predicate'))
```
Comment (misode) [inline java/1.20.5/src/schemas/Recipe.ts]: This should be `data_component_patch`, but I'm pretty sure you can replace this whole thing with `Reference('item_stack')`

File: java/1.20.5/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@mcschema/java-1.20.5",
-  "version": "0.0.6",
+  "version": "0.0.7",
```
Comment (misode) [inline java/1.20.5/package.json]: You don't need to bump this yet, I'll do it when I publish the npm package

File: java/1.20.5/src/schemas/index.ts [hunk 3]
```diff
@@ -26,6 +28,7 @@ export function initSchemas(schemas: SchemaRegistry, collections: CollectionRegi
     initChatTypeSchemas(schemas, collections)
     initAdvancementSchemas(schemas, collections)
     initAssetsSchemas(schemas, collections)
+    initBannerPatternSchema(schemas, collections)
```
Comment (misode) [inline java/1.20.5/src/schemas/index.ts]: Convention is plural `Schemas`, also below

File: java/1.20.5/src/schemas/WolfVariant.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import { Tag } from './Common'
+
+export function initWolfVariantSchema(schemas: SchemaRegistry, collections: CollectionRegistry) {
+  const StringNode = RawStringNode.bind(undefined, collections)
+
+	schemas.register('wolf_variant', Mod(ObjectNode({
+		texture: StringNode(),
+		angry_texture: StringNode(),
+		tame_texture: StringNode(),
+		biomes: Opt(Tag({ resource: '$worldgen/biome' })),
```
Comment (misode) [inline java/1.20.5/src/schemas/WolfVariant.ts]: Looking at the code, I don't think this field is optional, unless you tested this?

**Review by misode — COMMENTED** [review]
I didn't comment on Components yet

**Comment by Apollounknowndev** [comment]
That should be all of the component changes, just remembered the slot field was added so I'll work on that next and that should be everything

**Comment by Apollounknowndev** [comment]
Since there are so many possible slot values (161 total I think?), I currently have the keys for the `slot` field accept any string but have the collection slots (`container.*`, `enderchest.*`, etc.) suggested by default. Might not be the best solution, I can change it if needed.

**Review by misode — APPROVED** [review]
Thanks a lot for this!

**Pull request merged**